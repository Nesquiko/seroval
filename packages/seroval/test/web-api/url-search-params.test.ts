import { describe, it, expect } from 'vitest';
import {
  deserialize,
  fromJSON,
  serialize,
  serializeAsync,
  toJSON,
  toJSONAsync,
} from '../../src';

describe('URLSearchParams', () => {
  describe('serialize', () => {
    it('supports URLSearchParams', () => {
      const example = new URLSearchParams('hello=world&foo=bar');
      const result = serialize(example);
      expect(result).toMatchSnapshot();
      const back = deserialize<URLSearchParams>(result);
      expect(back).toBeInstanceOf(URLSearchParams);
      expect(String(back)).toBe(String(example));
    });
  });
  describe('serializeAsync', () => {
    it('supports URLSearchParams', async () => {
      const example = new URLSearchParams('hello=world&foo=bar');
      const result = await serializeAsync(Promise.resolve(example));
      expect(result).toMatchSnapshot();
      const back = await deserialize<Promise<URLSearchParams>>(result);
      expect(back).toBeInstanceOf(URLSearchParams);
      expect(String(back)).toBe(String(example));
    });
  });
  describe('toJSON', () => {
    it('supports URLSearchParams', () => {
      const example = new URLSearchParams('hello=world&foo=bar');
      const result = toJSON(example);
      expect(JSON.stringify(result)).toMatchSnapshot();
      const back = fromJSON<URLSearchParams>(result);
      expect(back).toBeInstanceOf(URLSearchParams);
      expect(String(back)).toBe(String(example));
    });
  });
  describe('toJSONAsync', () => {
    it('supports URLSearchParams', async () => {
      const example = new URLSearchParams('hello=world&foo=bar');
      const result = await toJSONAsync(Promise.resolve(example));
      expect(JSON.stringify(result)).toMatchSnapshot();
      const back = await fromJSON<Promise<URLSearchParams>>(result);
      expect(back).toBeInstanceOf(URLSearchParams);
      expect(String(back)).toBe(String(example));
    });
  });
});
