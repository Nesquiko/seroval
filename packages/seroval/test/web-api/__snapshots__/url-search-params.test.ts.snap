// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`URLSearchParams > crossSerialize > supports URLSearchParams 1`] = `"$R[0]=new URLSearchParams(\\"hello=world&foo=bar\\")"`;

exports[`URLSearchParams > crossSerializeAsync > supports URLSearchParams 1`] = `"$R[0]=Promise.resolve($R[1]=new URLSearchParams(\\"hello=world&foo=bar\\"))"`;

exports[`URLSearchParams > crossSerializeStream > supports URLSearchParams 1`] = `"$R[0]=_$.P()"`;

exports[`URLSearchParams > crossSerializeStream > supports URLSearchParams 2`] = `"_$.Ps($R[0],$R[1]=new URLSearchParams(\\"hello=world&foo=bar\\"))"`;

exports[`URLSearchParams > serialize > supports URLSearchParams 1`] = `"new URLSearchParams(\\"hello=world&foo=bar\\")"`;

exports[`URLSearchParams > serializeAsync > supports URLSearchParams 1`] = `"Promise.resolve(new URLSearchParams(\\"hello=world&foo=bar\\"))"`;

exports[`URLSearchParams > toJSON > supports URLSearchParams 1`] = `"{\\"t\\":{\\"t\\":19,\\"i\\":0,\\"s\\":\\"hello=world&foo=bar\\"},\\"f\\":16383,\\"m\\":[]}"`;

exports[`URLSearchParams > toJSONAsync > supports URLSearchParams 1`] = `"{\\"t\\":{\\"t\\":12,\\"i\\":0,\\"s\\":1,\\"f\\":{\\"t\\":19,\\"i\\":1,\\"s\\":\\"hello=world&foo=bar\\"}},\\"f\\":16383,\\"m\\":[]}"`;
